packages/ember-data/lib/system/model/model.js

```diff
@@ -907,11 +909,12 @@ var Model = Ember.Object.extend(Ember.Evented, {
     @method adapterDidCommit
   */
   adapterDidCommit: function(data) {
-    var changedKeys;
     set(this, 'isError', false);
 
+    var changedKeys = this._changedKeys(data);
+
     if (data) {
-      changedKeys = mergeAndReturnChangedKeys(this._data, data);
+      merge(this._data, data);
     } else {
```

Comment:
We should always merge inflight while we are at it, it's related to the other bug we looked at